### ISSUE TITLE
Allow creating directories when renaming files (behavior like create file)

### ISSUE THREAD

**Issue by SimonEggert** [nvim-tree/nvim-tree.lua#2630]
**Is your feature request related to a problem? Please describe.**
I recently discovered that it is currently not possible to rename a file and move it to another folder via nvim-tree's rename operation. I had hoped it worked like the way creating files works where you can use `foo/bar` to create a new folder `foo` with a `bar` file inside it. Rename just creates a file called `foo/bar` instead.

**Describe the solution you'd like**
When renaming a file `./bar` I want to be able to edit the "file name" to `./foo/bar` and have the file be nested in the created folder.

**Describe alternatives you've considered**
A colleague mentioned that VSCode had this feature when renaming files.

**Additional context**
The issue is not a big deal but I expected the two operations "rename" and "create" to be similar in behavior. Could this be done in a similar way in the code? I think the folder creation happens in [this loop in the code](https://github.com/nvim-tree/nvim-tree.lua/blob/f24afa2cef551122b8bd53bb2e4a7df42343ce2e/lua/nvim-tree/actions/fs/create-file.lua#L79..L109).

Thanks for considering and keep up the great work! 🙂

**Comment by alex-courtis** [comment]
This could be added to any similar operation we find.

**Comment by mohamedarish** [comment]
This could be implemented, but it also produces a problem of typos being enhanced

If a user wants to rename a file and make it go into a long nested directory.
If a typo is found in the root level directory of the new directory, then it shall result in the creation of many unnecessary nested directories and the initial file/folder being moved into this new directory tree.
Creating and then moving files does not create this problem.

**Comment by alex-courtis** [comment]
> If a typo is found in the root level directory of the new directory, then it shall result in the creation of many unnecessary nested directories and the initial file/folder being moved into this new directory tree.

Indeed. We can't really warn as it's not possible to determine the user's desired behaviour.

We can use `completion = "dir"` to assist the user when using existing directories.
